core/trino-spi/src/main/java/io/trino/spi/predicate/TupleDomain.java

```diff
@@ -296,10 +296,10 @@ public static <T> TupleDomain<T> intersect(List<? extends TupleDomain<? extends
     }
 
     @SuppressWarnings("unchecked")
-    private static <U, T extends U> TupleDomain<U> upcast(TupleDomain<T> domain)
+    private static <T> TupleDomain<T> upcast(TupleDomain<? extends T> domain)
```

Comment:
This is a correct simplification.
Was the original code an incorrect Java, according to JLS?

Reply:
The compiler complained about the usage:
```
The method upcast(TupleDomain<T>) in the type TupleDomain<T> is not applicable for the arguments (capture#20-of ? extends TupleDomain<? extends T>)
```

so it couldn't infer the U type, I'm not sure, if ECJ is too limited, or Javac is more lazy in this case.

Reply:
I am fine with this particular change. In the past, I've seen ECJ having problems where it shouldn't. I don't have any recent experience with this compiler.
Is there any reason you want to use ECJ?

Reply:
Yes, because this is what Eclipse is using,and a bit annoying that I can't start the app, without fixing the errors :wink:

Reply:
Isn't it possible to configure Eclipse to use javac as a compiler?